### ISSUE TITLE
Upgrade PHP version is not showing in Diagnostics GUI

### ISSUE THREAD

**Issue by maro584** [MISP/MISP#6449]
## Please consider the following notes
- Critical security bugs can be reported, preferably PGP encrypted, by mail to: info@circl.lu
- Bug reports and feature requests can be filed as an issue on github: https://github.com/MISP/MISP/issues
- For interactive support please join the Gitter chat on: https://gitter.im/MISP/MISP

The official documentation of MISP can be found here: https://www.circl.lu/doc/misp/
We also periodically do user/admin/developer trainings and have our training material online: https://www.circl.lu/services/misp-training-materials/

Nevertheless you can of course file a Support request as an issue. Please be as precise as possible and fill the template as detailed as possible too.
Please remove this text until the line below. Thanks a lot.
--------8<------

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Support
| OS version (server)       | RedHat
| OS version (client)       | 7
| PHP version               | 7.4
| MISP version / git hash   | 2.4.132
| Browser                   | If applicable

### Support Questions

I have upgraded a PHP from 7.1 to 7.4 . Using command "php -v" as Apache and Root user I got info that they used php 7.4 unfortunatelly in MSIP gui in diagnostic section I see that MISP is still using PHP 7.1 with forced path to it like:
PHP ini path:… /etc/opt/rh/rh-php71/php.ini

My question is, how to force MISP to use this path:

PHP ini path:… /etc/php.ini

I think it will solve a problem but any other ideas which helps will be appreciated.

### Logs, screenshots, configuration dump, ...

![image](https://user-images.githubusercontent.com/34789307/96259504-fe066680-0fbd-11eb-8664-64b49aeaa55d.png)

*******:Console# sudo -H -u apache php -v
PHP 7.4.11 (cli) (built: Sep 29 2020 10:17:06) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies

**Comment by JakubOnderka** [comment]
The problem is that in your system are probably installed two PHP interpreters. One from Software collections and second from system packages.

**Comment by maro584** [comment]
I have reinstall a php74 and install rh-php72  rh-php73, and to do that I used Software Collections. My rh direcetory looks like this:

/etc/opt/rh

rh-mariadb102  rh-php71  rh-php72  rh-php73  rh-python36  rh-redis32

PHP74 is installed in other location:

/etc/opt/remi/php74

Now when I am using php71 there is no issue with MISP GUI, but If I  am try to use other php version, then I got "Internal Server Error" in the GUI and also this log in: /var/www/MISP/app/tmp/logs/error.log:

```
2020-10-20 10:56:06 Error: [MissingConnectionException] Database connection "Mysql" is missing, or could not be created.
Exception Attributes: array (
  'class' => 'Mysql',
  'message' => 'SQLSTATE[HY000] [2002] No such file or directory',
  'enabled' => true,
)
Request URL: /
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(278): Mysql->connect()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/ConnectionManager.php(105): DboSource->__construct()
#2 /var/www/MISP/app/Controller/AppController.php(622): ConnectionManager::getDataSource()
#3 /var/www/MISP/app/Controller/AppController.php(148): AppController->_setupDatabaseConnection()
#4 /var/www/MISP/app/Controller/EventsController.php(62): AppController->beforeFilter()
#5 /var/www/MISP/app/Lib/cakephp/lib/Cake/Event/CakeEventManager.php(243): EventsController->beforeFilter()
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(682): CakeEventManager->dispatch()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(189): Controller->startupProcess()
#8 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#9 /var/www/MISP/app/webroot/index.php(92): Dispatcher->dispatch()
#10 {main}
```
I have installed mysql php library and redis php library.

I am starting fpm services like this:

rh-php71 systemctl start rh-php71-php-fpm.service
rh-php72 systemctl start rh-php72-php-fpm.service
rh-php73 systemctl start rh-php73-php-fpm.service
rh-php74 systemctl start php-fpm.service


handler is set in /etc/httpd/conf.d/misp.conf

<FilesMatch \.php$>
                        SetHandler "proxy:fcgi://127.0.0.1:9000"
                </FilesMatch>

**Comment by maro584** [comment]
I have found a solution for this issue.
I have removed rh-php72, rh-php73, rh-php74 and use this guide to upgrade php:
https://computingforgeeks.com/how-to-install-php-7-4-on-centos-7/
But still GUI was not displaying properly there was two main issues:
1. In file database.php I have change a word "localhost" to "127.0.0.1" it mitigate this error.log:
2020-10-20 10:56:06 Error: [MissingConnectionException] Database connection "Mysql" is missing, or could not be created.
Exception Attributes: array (
  'class' => 'Mysql',
  'message' => 'SQLSTATE[HY000] [2002] No such file or directory',
  'enabled' => true,
)
2. I have install php-redis library/extension to mitigate this log:
Error: [Error] Class 'Redis' not found

After all of that things I have restart php-fpm service and GUI is working correct.
